arduino/smartcar/smartcar.ino

```diff
@@ -158,9 +155,44 @@ void handleAngleTopic(int input){
     }
     car.update();
 }
+/**
+ * Method to control the car directly from the serial in the emulator.
+ */
+void handleInput() {
+    if (Serial.available()) {
+        String input = Serial.readStringUntil('\n');
+        //TODO: Look at how the mqtt com has been implemented and how it impacts this method
+        if (input.startsWith("s")) {
+            // front and back sensors and we look at the + or - for direction
+            int inputSpeed = input.substring(1).toInt();
+            if (inputSpeed > 0) {
+                float frontValue = frontIR.getDistance();
+                handleSpeedInput(frontValue, inputSpeed);
+            } else if (inputSpeed < 0) {
+                float backValue = backIR.getDistance();
+                handleSpeedInput(backValue, inputSpeed);
+            } else {
+                car.setSpeed(0);
+            }
+        } else if (input.startsWith("a")) {
+            // look at the angle + or - :  + -> right and - -> left
+            int inputAngle = input.substring(1).toInt();
+            if (inputAngle > 0) {
+                float rightValue = rightIR.getDistance();
+                handleAngleInput(rightValue, inputAngle);
+            } else if (inputAngle < 0) {//get left sensor
+                float leftValue = leftIR.getDistance();
+                handleAngleInput(leftValue, inputAngle);
+            } else {
+                car.setAngle(0);
+            }
+            car.update();
+        }
+    }
+}
 
 /**
- * handleInput helper method for speed
+ * helper for MQTT and Serial inputs taking care of the speed
  */
 void handleSpeedInput(float distance, int inputSpeed){
```

Comment:
in handleSpeedInput() if-statement, comparing int with float. possible data loss might cause/trigger some other inconstancy in reaction.